frontend/src/pages/EditProductPage.js

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Formik, Form } from 'formik'
+import { useParams, useHistory } from 'react-router-dom'
+import { Button } from 'react-bootstrap'
+import InputField from '../components/InputField'
+import SelectCategory from '../components/listofproducts/SelectCategory'
+import validationEdit from '../validations/validationEdit'
+import { updateProductData, fetchOneProduct } from '../communications/userApi'
+import ResponseModal from '../components/modals/ResponseModal'
+
+const updateOldFieldsInItem = (newProductValues, product) => {
+  Object.keys(product).forEach((key) => {
+    newProductValues[key] && (product[key] = newProductValues[key])
```

Comment:
Ez már jobb, de ez meg így egy éselésnek álcázott if:
```
if(newProductValues[key]) {
    product[key] = newProductValues[key]
}
```

A spreadingnek akkor is működnie kellene, csak ennyi lenne forEach nélkül a függvény:
`return {...product, ...newProductValues}`

Reply:
return { ...product, ...newProductValues } ---> ez a verzió a product-ot a newProductValues értékeivel tölti fel, amelyek között sok üres String szerepelhet, ergo BE-en elszáll validációkor
return { ...newProductValues, ...product }  ---> ez pedig értelemszerűen minden newProductValues-t felülír a régi product értékekkel
Én maradnék a logical AND-nél, ha nem bánod, vagy van másik ötleted?